api/app/controllers/spree/api/v2/storefront/order_status_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class OrderStatusController < ::Spree::Api::V2::BaseController
+          def show
+            render_serialized_payload { serialize_resource(resource) }
+          end
+
+          private
+
+          def resource
+            resource = dependencies[:resource_finder].new(number: params[:number], token: request.headers['X-Spree-Order-Token']).execute.take
```

Comment:
@przemosk and let's use `order_token` method here instead of `request.headers['X-Spree-Order-Token']`